src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4755,7 +4810,9 @@ public static bool ConnectAsync(SocketType socketType, ProtocolType protocolType
                 MultipleConnectAsync multipleConnectAsync = null;
                 if (dnsEP.AddressFamily == AddressFamily.Unspecified)
                 {
-                    multipleConnectAsync = new MultipleSocketMultipleConnectAsync(socketType, protocolType);
+                    multipleConnectAsync = SocketPal.SupportsMultipleConnectAttempts ?
+                        (MultipleConnectAsync)(new DualSocketMultipleConnectAsync(socketType, protocolType)) :
```

Comment:
I see that the the following has been renamed but I don't understand the reason:
`   internal class MultipleSocketMultipleConnectAsync : MultipleConnectAsync`
`+    internal sealed class DualSocketMultipleConnectAsync : MultipleConnectAsync`